.github/workflows/api-compatibility.yml

```diff
@@ -0,0 +1,127 @@
+
+            # Install Rust toolchain on Windows for cargo prebuild
+            # - name: Install Rust (Windows)
+            #   shell: pwsh
+            #   run: |
+            #       $ProgressPreference = 'SilentlyContinue'
+            #       Invoke-WebRequest -Uri https://win.rustup.rs/x86_64 -OutFile rustup-init.exe
+            #       ./rustup-init.exe -y
+            #       echo "$env:USERPROFILE\.cargo\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
+            #       rustc -V
+            #       cargo -V
+
+            - name: Install dotnet-api-diff (Chocolatey)
+              shell: pwsh
+              run: |
+                  choco install dotnetapidiff --version=1.0.0 -y
+                  dotnetapidiff --version
+
+            # - name: Restore dependencies (for tooling)
+            #   shell: pwsh
+            #   run: |
+            #       dotnet restore benchmarks/csharp_benchmark.csproj
+
+            - name: Download StackExchange.Redis package (${{ env.SER_VERSION }})
+              shell: pwsh
+              run: |
+                  nuget install StackExchange.Redis -Version "$env:SER_VERSION" -OutputDirectory ./packages
+
+            - name: Build Valkey.Glide (net8.0)
+              shell: pwsh
+              run: |
+                  dotnet build --configuration Release --framework net8.0 --nologo
+
+            - name: Run API Diff (net8.0)
+              id: run-diff
+              shell: bash
+              run: |
+                  set -euo pipefail
+                  mkdir -p api-diff
+
+                  VALKEY_DLL="sources/Valkey.Glide/bin/Release/net8.0/Valkey.Glide.dll"
+                  SER_DLL="packages/StackExchange.Redis.${SER_VERSION}/lib/net8.0/StackExchange.Redis.dll"
+                  OUTFILE="api-diff/api-changes-net8.0.html"
+
+                  echo "Using StackExchange.Redis version: ${SER_VERSION}"
+                  echo "SER DLL: ${SER_DLL}"
+                  echo "Valkey DLL: ${VALKEY_DLL}"
+
+                  if [[ ! -f "$SER_DLL" ]]; then
+                    echo "# API Diff Error (net8.0)" > "$OUTFILE"
+                    echo "Could not find StackExchange.Redis DLL at $SER_DLL" >> "$OUTFILE"
+                    echo "exit_code=6" >> "$GITHUB_OUTPUT"
+                    exit 0
+                  fi
+                  if [[ ! -f "$VALKEY_DLL" ]]; then
+                    echo "# API Diff Error (net8.0)" > "$OUTFILE"
+                    echo "Could not find Valkey.Glide DLL at $VALKEY_DLL" >> "$OUTFILE"
+                    echo "exit_code=6" >> "$GITHUB_OUTPUT"
+                    exit 0
+                  fi
+
+                  set +e
+                  dotnetapidiff compare \
+                    "$SER_DLL" \
+                    "$VALKEY_DLL" \
+                    --config ".github/api-compat/api-diff-config.json" \
+                    --output html \
+                    --output-file "$OUTFILE"
+                  CODE=$?
```

Comment:
if you remove `set +e`, GHA will fail the pipeline if exit code is not 0. So you can remove lines 102-106 and the last step too